entity-framework/core/modeling/owned-entities.md

```diff
@@ -0,0 +1,81 @@
+
+The OwnsOne fluent mapping alerts the DbContext to this mapping. In this example, StreetAddress is a type with no identity property. It is used as a property of the Order type to specify the shipping address for a particular order. In the DbContext, you can use the Entity().OwnsOne() method to specify that the ShippingAddress property is an Owned Entity of the Order type.
+
+```
+public class StreetAddress
+{
+    public string Street { get; set; }
+    public string City { get; set; }
+}
+
+public class Order
+{
+    public int Id { get; set; }
+    public StreetAddress ShippingAddress { get; set; }
+}
+
+modelBuilder.Entity<Order>().OwnsOne(p => p.ShippingAddress);
+```
+If the ShippingAddress property is private in the Order type, you can use an overloaded signature of the OwnsOne method:
+
+```
+modelBuilder.Entity<Order>().OwnsOne(typeof(StreetAddress), "ShippingAddress");
+```
+
+EF Core convention will name name the database columns for the properties of the owned entity type as EntityProperty_OwnedEntityProperty. Therefore the StreetAddress properties will appear in the Orders table with the names ShippingAddress_Street and ShippingAddress_City. You can append the Property()HasColumnName() fluent method to rename those columns. In the case where StreetAddress is a public property, the mappings would be
+
+```
+modelBuilder.Entity<Order>().OwnsOne(p => p.ShippingAddress)
+                            .Property(p=>p.Street).HasColumnName("ShipsToStreet");
+modelBuilder.Entity<Order>().OwnsOne(p => p.ShippingAddress)
+                            .Property(p=>p.City).HasColumnName("ShipsToCity");
+```
+In the case of the private ShippingAddress property, you need only to replace the Property lambda expressions with strings.
+
+```
+modelBuilder.Entity<Order>().OwnsOne(typeof(StreetAddress), "ShippingAddress")
+                            .Property("Street").HasColumnName("ShipsToStreet");
+modelBuilder.Entity<Order>().OwnsOne(typeof(StreetAddress), "ShippingAddress")
+                            .Property("City").HasColumnName("ShipsToCity");
+```
+
+It is possible to chain the OwnsOne method in a fluent mapping. In this example OrderDetails owns BillingAddress and ShippingAddress, which are both StreetAddress types. Then OrderDetails is owned by the Order type.
+
+``` csharp
+modelBuilder.Entity<Order>().OwnsOne(p => p.OrderDetails, cb =>
+    {
+        cb.OwnsOne(c => c.BillingAddress);
+        cb.OwnsOne(c => c.ShippingAddress);
+    });
+
+public class Order
+{
+    public int Id { get; set; }
+    public OrderDetails OrderDetails { get; set; }
+}
+
+public class OrderDetails
+{
+    public StreetAddress BillingAddress { get; set; }
+    public StreetAddress ShippingAddress { get; set; }
+}
+
+public class StreetAddress
+{
+    public string Street { get; set; }
+    public string City { get; set; }
+}
+```
```

Comment:
We should also call out the restrictions on owned types in 2.0:

- They don't support inheritance
- They cannot be pointed at by a collection navigation property
- Since they use table splitting by default owned types also have the following restrictions unless explicitly mapped to a different table:
  - They cannot be owned by a derived type
  - The navigation property cannot be `null`

Reply:
Agreed. There is some more info in https://github.com/aspnet/EntityFramework.Docs/issues/466#issuecomment-326791656, but we should filter it and keep only what is most helpful.